ethcore/engines/validator-set/src/contract.rs

```diff
@@ -61,13 +65,34 @@ impl ValidatorContract {
 }
 
 impl ValidatorContract {
-	fn transact(&self, data: Bytes) -> Result<(), String> {
+	fn transact(&self, data: Bytes, reporting: &mut ValidatorReporting, sender: Address) -> Result<(), String> {
 		let client = self.client.read().as_ref()
 			.and_then(Weak::upgrade)
 			.ok_or_else(|| "No client!")?;
 
 		match client.as_full_client() {
 			Some(c) => {
+				let latest_block_num = c.block_number(BlockId::Latest).expect("Latest block is never None");
+				match reporting.config {
+					ReportingConfig::Disable => return Err("Reporting is disabled.".into()),
+					ReportingConfig::Force => {},
+					ReportingConfig::Call(blk_num) => {
+						let checked_last_block = reporting.checked_last_block;
+						if checked_last_block.is_none() || (latest_block_num - checked_last_block.expect("checked for None before")) >= blk_num {
```

Comment:
Maybe easier to read?
```suggestion
						if latest_block_num - checked_last_block.unwrap_or(0) >= blk_num {
```